Run nativeaot tests in helix

sbomer opened the pull request:
This changes the existing NativeAot smoke test runs to run in helix. I changed the multi-module test run (only supported on windows) to run only in the checked configuration, to avoid having to add a separate windows build script invocation inside one of the yaml templates.

sbomer committed:
Run nativeaot tests in helix

sbomer committed:
Fix timeout

sbomer committed:
Remove condition

sbomer committed:
Fix test parameters

agocke commented:
Hm, not sure how this works. Weren't we already running the smoke tests on the build machine? Does this disable that run? Or just add a new one?

agocke commented:
(nvm didn't notice this was a draft)

sbomer committed:
More fixes

- buildConfig casing
- Support uniform test filter syntax for bash/cmd
- Remove problematic quotes from testFilter argument

sbomer committed:
Use nativeaot test launcher script

And remove broken os arg.

Also clean up:
- release config comment
- use of platform-specific test filter

sbomer committed:
More fixes

- Pass args not recognized by build.cmd last
- Make native exe executable on unix

sbomer committed:
Factor existing job definitions

- Replace existing nativeaot smoke test jobs with
  helix versions
- Move BuildNativeAotFrameworkObjects out to runtime.yml, and only pass
it for checked windows job.
- Use nativeaot-post-build-steps to continue building
  nativeaot test apps

sbomer committed:
Pass through parameters to inner template

sbomer committed:
Add helixQueues parameter pass-through

sbomer committed:
Don't pass build config as libraries config

Also clean up name suffixes

sbomer committed:
Try running pri0 release tests

sbomer committed:
Revert "Try running pri0 release tests"

This reverts commit 967c7ace837d07c2edfb03ef6b32d74d33f850f6.

sbomer committed:
Use buildconfig as suffix

trylek commented on eng/pipelines/common/templates/runtimes/build-runtime-tests-and-send-to-helix.yml:
```diff
@@ -26,11 +27,11 @@ parameters:
 
   - ${{ if ne(parameters.runtimeFlavor, 'mono') }}:
-    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) /p:LibrariesConfiguration=${{ parameters.buildConfig }} -ci os ${{ parameters.osGroup }} ${{ parameters.archType }} $(buildConfigUpper)
+    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) ${{ parameters.archType }} $(buildConfigUpper) ${{ parameters.testBuildArgs }} -ci
```
Why don't we need the LibrariesConfiguration in this case?

trylek reviewed (approved):
LGTM, thank you!

MichalStrehovsky commented on eng/pipelines/runtime.yml:
```diff
@@ -240,17 +266,30 @@ extends:
             testGroup: innerloop
             timeoutInMinutes: 120
-            nameSuffix: NativeAOT
+            nameSuffix: NativeAOT_Helix
```
Why do we need to change the suffix here?

MichalStrehovsky reviewed (approved):
Thank you!

sbomer committed:
PR feedback

- Remove unnecessary parameters
- Clean up nameSuffix
